TTlibrary/app/controllers/books_controller.rb

```diff
@@ -15,18 +15,25 @@ def index
     @books = author.books
   end
 
-  # GET /books/1
-  # GET /books/1.json
+  # GET author/:author_id/books/:id
+  # GET author/:author_id/books/:id.json
   def show
     @comments = @book.comments
+    if user_signed_in?
+      @request = Request.where('book_id' => params[:id], 'user_id' => current_user.id)
```

Comment:
Just a comment:
Instead of `@request = Request.where('book_id' => params[:id], 'user_id' => current_user.id)` you can do `@book_already_requested = current_user.books.pluck(:id).include?(id)`.

It may be a more "natural" way to code it (or not), but the result is the same.